app/src/main/java/com/example/grutor/Adapters/LessonAdapter.java

```diff
@@ -3,43 +3,66 @@
 import android.annotation.SuppressLint;
 import android.content.Context;
 import android.content.Intent;
-import android.os.Bundle;
+import android.database.DataSetObserver;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.graphics.drawable.Drawable;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.EditText;
+import android.widget.ImageButton;
 import android.widget.ImageView;
 import android.widget.TextView;
-import android.widget.Toast;
 
 import androidx.annotation.NonNull;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.ItemTouchHelper;
 import androidx.recyclerview.widget.RecyclerView;
 
+import com.example.grutor.Activites.FeedActivity;
+import com.example.grutor.Fragments.MessagesFragment;
+import com.example.grutor.Modals.Groupchat;
 import com.example.grutor.Modals.Lessons;
 import com.example.grutor.R;
+import com.example.grutor.Utility.studentMatcher;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.snackbar.Snackbar;
+import com.parse.ParseException;
 import com.parse.ParseUser;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class LessonAdapter extends RecyclerView.Adapter<LessonAdapter.ViewHolder> {
-    List<Lessons> lessons;
-    LayoutInflater inflater;
-    Context context;
+    public List<Lessons> lessons;
+    protected LayoutInflater inflater;
+    protected Context context;
+    public String requestedLessonString;
+    public Lessons requestedLesson;
 
     public class ViewHolder extends RecyclerView.ViewHolder {
 
         public TextView tvNumType, tvDateTime;
         public EditText etBundledDescription;
         public ImageView ivSubjectLesson;
+        protected FloatingActionButton fabChat;
         public Button btnSubjectTopic;
+        public boolean clicky;
```

Comment:
the view item and viewholder will be recycling used, define clicky in ViewHolder can't hold it if recycler view scrolling. you should use another way to set the flag after item is clilcked.